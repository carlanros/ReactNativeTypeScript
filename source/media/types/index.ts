// media/types/index.ts

export enum MediaConstants {
  /**
   * Login failed or was cancelled
   */
  MEDIA_FAILED = 'MEDIA_FAILED',
  /**
   * Login state: authentication succeeded
   *
   * @static
   * @memberof MediaConstants
   */
  MEDIA_SUCCESS = 'MEDIA_SUCCESS',

  /**
   * Login state: authentication in progress
   *
   * @static
   * @memberof MediaConstants
   */
  MEDIA_BUSY = 'Kollar vem du Ã¤r',
}

export interface IMediaState {
  cancelled: boolean;
  height: number;
  width: number;
  uri: string;

}
export interface IMediaAction extends IMediaState {
  type: MediaConstants;
}
