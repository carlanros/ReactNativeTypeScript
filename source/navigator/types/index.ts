export interface INavigationState {
  /**
   * Current scene object in navigation
   *
   * @type {object}
   * @memberof INavigationReducer
   */
  scene: object;
}

export interface INavigationAction extends INavigationState {
  type: string;
}
