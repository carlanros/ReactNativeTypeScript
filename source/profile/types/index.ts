// profile/types/index
import * as types from '../../Types';

export interface IProfileProps extends types.IProps {
  Logout?: () => any;
}
