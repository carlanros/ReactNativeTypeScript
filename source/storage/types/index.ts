// storage/types/index

/**
 * App storage Redux action types
 * @type {StorageConstants}
 * @public
 * @type {Object}
 */
export enum StorageConstants {
    STORAGE_GOTKEY = 'STORAGE = GOT KEY',
    STORAGE_REMOVEDKEY = 'STORAGE = REMOVED KEY',
    STORAGE_REMOVED_FAILED = 'STORAGE = REMOVED KEY FAILED',
    STORAGE_NOSUCHKEY = 'STORAGE = NO SUCH KEY',
    STORAGE_SAVE_FAILED = 'STORAGE = SAVE FAILED',
    STORAGE_SAVED = 'STORAGE = SAVED KEY',
    STORAGE_USER_KEY = 'STORAGE KEY = USER',
    STORAGE_BUSY = 'Friskar upp minnet',
  }

export interface IKey {
/**
 * The key in app storage
 *
 * @type {string}
 */
  key: string;
}

/**
 * Reducer properties passed down from StorageReducer
 * @typedef IStorageReducer
 * @type {Object}
 * @public
 * @export
 * @property {Object} value - the app storage value
 * @property {string} key - The app storage key
 */
export interface IStorageState extends IKey {

  /**
   * Value to save / load from app storage
   *
   * @type {object?}
   */
  value?: object;
}

export interface IStorageAction extends IStorageState {
  type: StorageConstants;
}