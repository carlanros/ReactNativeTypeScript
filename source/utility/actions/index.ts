// utility/actions/index
import * as Redux from 'redux';
import * as appTypes from '../types';

/**
 * The app loading state has changed
 *
 * @function AppLoadingChanged
 * @memberof AppActions
 * @public
 * @global
 */
export const AppLoadingChanged = ({ isBusy, reason = null, sender = null }: appTypes.IUtilityLoadingChangedPayload) =>
  (dispatch: Redux.Dispatch<appTypes.IAppLoadingChanged>) => {
  dispatch({
    type: isBusy ? appTypes.UtilityConstants.APP_LOAD_BUSY : appTypes.UtilityConstants.APP_LOAD_DONE,
    reason,
    sender,
  });
};

export type AppLoadingChanged = typeof AppLoadingChanged;

/**
 * The app error state has changed
 *
 * @function AppErrorChanged
 * @public
 * @global
 */
export const AppErrorChanged = ({
  hasError, reason = null, exception = null, sender = null,
}: appTypes.IUtilityErrorPayload) =>
  (dispatch: Redux.Dispatch<appTypes.IAppErrorChanged>) => {
    dispatch({
      type: hasError ? appTypes.UtilityConstants.APP_HAS_ERROR : appTypes.UtilityConstants.APP_NO_ERROR,
      reason,
      exception,
      sender,
    });
  };

  export type AppErrorChanged = typeof AppErrorChanged;