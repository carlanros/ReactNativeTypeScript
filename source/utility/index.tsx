// utility/index
import UtilityReducer from './reducers';
import * as UtilityTypes from './types';
import BusyIndicator, {
  IBusyIndicatorProps,
  IndicatorType,
  Spinner,
} from './components/BusyIndicator';
import { AppErrorChanged, AppLoadingChanged } from './actions';

export {
  AppErrorChanged,
  AppLoadingChanged,
  UtilityReducer,
  UtilityTypes,
  BusyIndicator,
  IBusyIndicatorProps,
  IndicatorType,
  Spinner,
};
