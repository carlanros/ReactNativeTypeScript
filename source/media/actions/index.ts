// media/actions/intex.ts
import * as Redux from 'redux';
import axios from 'axios';
import { Constants, Facebook } from 'expo';
import * as mediaTypes from '../types/';
import { AppErrorChanged, AppLoadingChanged } from '../../utility/';
import { StorageTypes, SaveByKey } from '../../storage';
// import { MediaConstants } from '../types/index';

const takePhoto = 'TakePhoto';
/**
 * Login to Facebook and store user to AsyncStorage
 * @function FacebookLogin
 * @returns {UserDispatch}
 * @memberof FacebookActions
 * @public
 * @global
 */
export const FacebookLogin = () => async (
  dispatch: Redux.Dispatch<mediaTypes.IMediaAction>
) => {
  dispatch(
    AppLoadingChanged({
      isBusy: true,
      reason: mediaTypes.MediaConstants.MEDIA_BUSY,
      sender: takePhoto,
    })
  );

  try {
// Expo bild:

      let actionResult: mediaTypes.IMediaAction = {
          type: mediaTypes.MediaConstants.MEDIA_SUCCESS,

      };

      if (!actionResult.cancelled) {
      dispatch(actionResult);
         // Reset errors
   dispatch(
       AppErrorChanged({ hasError: false, sender: takePhoto }));
    } else {

    dispatch(
        AppErrorChanged({
          hasError: true,
          reason: 'Photo was cancelled',
          exception: mediaTypes.MediaConstants.MEDIA_FAILED,
          sender: takePhoto,
        })
      );
    }
     } catch (error) {
    dispatch(
      AppErrorChanged({
        hasError: true,
        reason: mediaTypes.MediaConstants.MEDIA_FAILED,
        exception: error,
        sender: takePhoto,
      })
    );
  } finally {
    dispatch(AppLoadingChanged({ isBusy: false, sender: takePhoto }));

  }
};