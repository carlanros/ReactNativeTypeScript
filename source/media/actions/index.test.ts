// action/index.test.ts
// @ts-ignore
import configureMockStore, { MockStore } from 'redux-mock-store';
import { Store } from '../../store';
import { TakePhoto } from '../actions';
import * as MediaTypes from '../types';
import { UtilityTypes } from '../../utility';
import { IMediaAction } from '../types/index';

const storeMock = configureMockStore<Store>();

describe('Media actions', () => {
  beforeEach(() => {
    // @ts-ignore
    storeMock.clearActions();
  });
  test('TakePhoto returns Success when done', async () => {
    const expectedPayload: MediaTypes.IMediaAction = {
      type: MediaTypes.MediaConstants.MEDIA_SUCCESS,
      cancelled: false,
      uri: 'test',
      width: 1,
      height: 1,
    };

    const AppLoading = (sender: string, reason: string) => {
      return {
        sender,
        reason,
        type: UtilityTypes.UtilityConstants.APP_LOAD_BUSY,
      };
    };

    const AppLoadDone = (sender: string, reason: string = null) => {
      return {
        sender,
        type: UtilityTypes.UtilityConstants.APP_LOAD_DONE,
        reason,
      };
    };
    const NoError = (
      sender: string,
      reason: string = null,
      exception: string = null
    ) => {
      return {
        sender,
        type: UtilityTypes.UtilityConstants.APP_NO_ERROR,
        reason,
        exception,
      };
    };
    const expectedActions = [
      AppLoading('TakePhoto', MediaTypes.MediaConstants.MEDIA_BUSY),
      expectedPayload,
      NoError('TakePhoto'),
      AppLoadDone('TakePhoto'),
    ];

    // Dispatch action
    // @ts-ignore
    await storeMock.dispatch(TakePhoto());

    // @ts-ignore
    expect(storeMock.getActions()).toMatchSnapshot();

    // @ts-ignore
    expect(storeMock.getActions()).toEqual(expectedActions);
  });
});
