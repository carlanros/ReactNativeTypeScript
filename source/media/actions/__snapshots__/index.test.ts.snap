// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media actions TakePhoto returns Success when done 1`] = `
Array [
  Object {
    "reason": "Kollar vem du Ã¤r",
    "sender": "TakePhoto",
    "type": "APP = BUSY LOADING",
  },
  Object {
    "cancelled": false,
    "height": 1,
    "type": "MEDIA_SUCCESS",
    "uri": "test",
    "width": 1,
  },
  Object {
    "exception": null,
    "reason": null,
    "sender": "TakePhoto",
    "type": "APP NO ERROR",
  },
  Object {
    "reason": null,
    "sender": "TakePhoto",
    "type": "APP = DONE LOADING",
  },
]
`;
